src/ophyd_async/epics/pvi/pvi.py

```diff
@@ -80,10 +82,15 @@ def _verify_common_blocks(entry: PVIEntry, common_device: Type[Device]):
         if sub_name in ("_name", "parent"):
             continue
         assert entry.sub_entries
-        if sub_name not in entry.sub_entries and get_origin(sub_device) is not Optional:
-            raise RuntimeError(
-                f"sub device `{sub_name}:{type(sub_device)}` was not provided by pvi"
-            )
+        device_t, is_optional = _strip_union(sub_device)
+        if sub_name not in entry.sub_entries:
+            if is_optional:
+                continue
+            else:
+                raise RuntimeError(
+                    f"sub device `{sub_name}:{type(sub_device)}` "
+                    "was not provided by pvi"
+                )
```

Comment:
```suggestion
        if sub_name not in entry.sub_entries and not is_optional:
            raise RuntimeError(
                f"sub device `{sub_name}:{type(sub_device)}` "
                "was not provided by pvi"
            )
```

Reply:
I couldn't do this without adding another `sub_name not in entry.sub_entries and is_optional`, else it will hit one of the other conditions below?

Reply:
Makes sense